client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,351 @@
+                ]
+            if key == "group" and default_value not in cls.group_enum_values:
+                default_value = ""
+            default_values[key] = default_value
+
+        attr_defs = [
+            NumberDef(
+                "chunk_size",
+                label="Frames Per Task",
+                default=default_values.get("chunk_size"),
+                decimals=0,
+                minimum=1,
+                maximum=1000
+            ),
+            NumberDef(
+                "priority",
+                label="Priority",
+                default=default_values.get("priority"),
+                decimals=0
+            ),
+            TextDef(
+                "department",
+                label="Department",
+                default=default_values.get("department")
+            ),
+            EnumDef(
+                "group",
+                label="Group",
+                default=default_values.get("group"),
+                items=cls.group_enum_values,
+            ),
+            EnumDef(
+                "limit_groups",
+                label="Limit Groups",
+                multiselection=True,
+                default=default_values.get("limit_groups"),
+                items=cls.limit_group_enum_values,
+            ),
+            EnumDef(
+                "primary_pool",
+                label="Primary pool",
+                default="none",
+                items=cls.pool_enum_values,
+            ),
+            EnumDef(
+                "secondary_pool",
+                label="Secondary pool",
+                default="none",
+                items=cls.pool_enum_values,
+            ),
+            EnumDef(
+                "machine_list",
+                label="Machine list",
+                multiselection=True,
+                default=default_values.get("machine_list"),
+                items=cls.machines_enum_values,
+            ),
+            BoolDef(
+                "machine_list_deny",
+                label="Machine List is a Deny",
+                default=default_values.get("machine_list_deny")
+            ),
+            TextDef(
+                "job_delay",
+                label="Delay job",
+                default=default_values.get("job_delay"),
+                tooltip="Delay job by specified timecode. Format: dd:hh:mm:ss",
+                placeholder="00:00:00:00"
+            )
+        ]
+
+        return [
+            attr_def
+            for attr_def in attr_defs
+            if attr_def.key in overrides
+        ]
+
+    @classmethod
+    def register_create_context_callbacks(cls, create_context):
+        create_context.add_value_changed_callback(cls.on_values_changed)
+
+    @classmethod
+    def on_values_changed(cls, event):
+        for instance_change in event["changes"]:
+            instance = instance_change["instance"]
+            #recalculate only if context changes
+            if "productName" not in instance_change:
+                continue
```

Comment:
I think you  care about context change, not product name change, or?
```suggestion
            if (
                "task" not in instance_change
                and "folderPath" not in instance_change
            ):
                continue
```